app/slimming/slimming.js

```diff
@@ -156,6 +158,7 @@ app.controller('GOSlimCtrl', function($scope, $location, $q,
     delete $scope.selection[termToRemove.aspect].terms[termToRemove.id];
     // Add to de-selected items
     $scope.deSelectedItems.push(termToRemove);
+    $scope.testIfSelectionIsEmpty();
```

Comment:
No need of a watch at all. I saw it is used to enable/disable the gene product id and taxon id options but a method should work as well, should not it? Please revise the method approach and if it works for it. 

In any case the method testIfSelectionIsEmpty is not needed.